keps/672-execution-policy/README.md

```diff
@@ -0,0 +1,401 @@
+desired replicated Jobs reach the status defined in replicatedJobsStatus, the controller creates
+the next set of replicated Jobs.
+
+If JobSet is suspended the all ReplicatedJobs will be suspended and the Job sequence starts again.
+
+### Quota Management
+
+In the initial implementation of the ExecutionPolicy the resource quota will be calculated as
+sum of all ReplicatedJobs resources. Which means JobSet will be admitted by
+[Kueue](https://github.com/kubernetes-sigs/kueue) only when all resources are available for
+every ReplicatedJob within JobSet.
+
+That allows us to leverage the existing integration between Kueue and JobSet while using the
+ExecutionPolicy API.
+
+In the future versions we will discuss potential partial admission of JobSet by Kueue. For example,
+when compute resources are available for the first ReplicatedJob the JobSet can be dispatched by
+Kueue.
+
+### Defaulting/Validation
+
+- ExecutionPolicy is immutable.
+- ExecutionPolicyOrder of `AnyOrder` is default setting.
+- StartupPolicy should be equal to `AnyOrder` when ExecutionPolicy is used.
+  - The StartupPolicy API will be deprecated over the next few JobSet releases, since
+    ExecutionPolicy can be used with ready status of ReplicatedJobs.
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+#### Unit Tests
+
+- `controllers`: `01/19/2024` - `30.2%`
+
+We will create a new file called execution_policy for the functionality.
+
+#### Integration tests
+
+- Having 2 Jobs running in sequence with Succeeded condition.
+- Suspend JobSet with execution policy being set.
+- Restart the failed Job when execution policy is set in JobSet.
+
+### Graduation Criteria
+
+<!--
+
+Clearly define what it means for the feature to be implemented and
+considered stable.
+
+If the feature you are introducing has high complexity, consider adding graduation
+milestones with these graduation criteria:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Feature gate][feature gate] lifecycle
+- [Deprecation policy][deprecation-policy]
+
+[feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+-->
+
+## Implementation History
+
+- Draft KEP: September 25th 2024
+
+## Drawbacks
+
+### Manage quota for Job sequence
+
+It is hard to manage quota for sequence of Job within single JobSet. Usually, users don't want to
+lock resources (e.g. GPUs/TPUs) when the first Job is running.
+
+### Support complex DAGs with JobSet
+
+If users want to create complex DAG workflows, they should not use JobSet for such purpose.
+
+## Alternatives
```

Comment:
I still would like to hear why moving this type of workflow to a designated workflow engine is not a valid option?

Why would one want to use ExecutionPolicy without using a dag workflow?

For Kubeflow, why would not folding this kind of workflow into a Kubeflow pipeline not be sufficient for your purpose?

TrainingOperator is used for training while prep or post work could be specified as part of a workflow engine.

It is also not clear to me why one would not want to use this to for data prep-blend (cleaning data, transforming data) which to me is really when you should use a workflow engine..

Reply:
Let me add some points about it in the alternative section.

Reply:
I've added my initial thoughts on it. From my point of view, as long as we suggest users to leverage this functionality to just initialization and preparing data for training, we should be fine.